app/models/setting.rb

```diff
@@ -287,6 +291,14 @@ def self.column_check(opts)
 
   private
 
+  def validate_host_owner
+    owner_type_and_id = self.value
+    return if owner_type_and_id.blank?
+    owner = User.find_by_id(owner_type_and_id.to_i) if owner_type_and_id =~ (/-Users\Z/)
```

Comment:
This code (or similar) is repeated quite a lot. What do you think of moving this to `UserFactory` class that you could use like this:

```
owner = UserFactory.new(owner_type_and_id).user_or_usergroup
```

Feel free to come up with a better name ;-)
An alternative might be to make this an instance method of the `HostOwnerSuggestion` service.